src/hooks/use-aggrid-sort.ts

```diff
@@ -26,29 +28,33 @@ export enum SortWay {
 }
 
 export const useAgGridSort = (
-    initSortConfig: SortConfigType
+    initSortConfig: SortConfigType[],
+    sortAction?: (tab: string, sortConfig: SortConfigType[]) => AnyAction,
+    tab?: string
 ): SortPropsType => {
-    const [sortConfig, setSortConfig] = useState<SortConfigType[]>([
-        initSortConfig,
-    ]);
-
-    const onSortChanged = useCallback((newSortConfig: SortConfigType) => {
-        setSortConfig((prevSortConfig) =>
-            prevSortConfig
-                // for now, we can have only one parent sort and one children sort
+    const [sortConfig, setSortConfig] =
+        useState<SortConfigType[]>(initSortConfig);
+    const dispatch = useDispatch();
+
+    const onSortChanged = useCallback(
+        (newSortConfig: SortConfigType) => {
+            const updatedSortConfig = sortConfig
                 .filter(
                     (sort) =>
                         (sort.children ?? false) !==
                         (newSortConfig.children ?? false)
                 )
-                .concat(newSortConfig)
-        );
-    }, []);
+                .concat(newSortConfig);
+
+            setSortConfig(updatedSortConfig);
```

Comment:
the previous way to update state was correct, that's how we should implement it when the state depends on the previous state value

plus, I'm a bit confused, why do we need "sortConfig" useState while everything is now stored in redux

Reply:
i removed the useState